api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts

```diff
@@ -0,0 +1,46 @@
+import { isDefined } from '@marxan/utils';
+import { FeatureCollection, GeoJSON, Geometry } from 'geojson';
+import { PlanningUnitCost } from '../ports/planning-unit-cost';
+import { PuExtractorPort } from '../ports/pu-extractor/pu-extractor.port';
+
+type Properties = {
+  cost: number;
+  planningUnitId: string;
+};
+type MaybeProperties = Partial<Properties> | undefined | null;
```

Comment:
this is quite useful in general - I suggest to create a `MaybeSome<T>` and then `type MaybeProperties = MaybeSome<Properties>` or something like that.

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/219/commits/573622f92a11fdd126982184dc16a4ab3e03ecb3

Reply:
https://github.com/piotrwitek/utility-types#nullish